qiskit/transpiler/passes/synthesis/unitary_synthesis.py

```diff
@@ -57,21 +61,63 @@ def _choose_euler_basis(basis_gates):
 class UnitarySynthesis(TransformationPass):
     """Synthesize gates according to their basis gates."""
 
-    def __init__(self, basis_gates: List[str], approximation_degree: float = 1):
-        """
-        Synthesize unitaries over some basis gates.
-
-        This pass can approximate 2-qubit unitaries given some approximation
-        closeness measure (expressed as approximation_degree). Other unitaries
-        are synthesized exactly.
+    def __init__(
+        self,
+        basis_gates: List[str],
+        approximation_degree: float = 1,
+        coupling_map: List = None,
+        backend_props: BackendProperties = None,
+        pulse_optimize: Union[bool, None] = None,
+        natural_direction: Union[bool, None] = None,
+        synth_gates: Union[List[str], None] = None,
+    ):
+        """Synthesize unitaries over some basis gates.
+
+        This pass can approximate 2-qubit unitaries given some
+        approximation closeness measure (expressed as
+        approximation_degree). Other unitaries are synthesized
+        exactly.
 
         Args:
             basis_gates: List of gate names to target.
-            approximation_degree: closeness of approximation (0: lowest, 1: highest).
+            approximation_degree: Closeness of approximation
+                (0: lowest, 1: highest).
+            backend_props: Properties of a backend to synthesize for
+                (e.g. gate fidelities).
+            pulse_optimize: Whether to optimize pulses during
+                synthesis. A value of None will attempt it but fall
+                back if it doesn't succeed.
+            natural_direction: Whether to apply synthesis considering
+                directionality of 2-qubit gates. Only applies when
+                `pulse_optimize` == True. The natural direction is
+                determined by first checking to see whether the
+                coupling map is unidirectional.  If there is no
+                coupling map, the gate direction with the shorter
+                duration from the backend properties will be used. If
```

Comment:
Also this docstring for `natural_direction` should be updated to reflect the new behavior after the changes Matthew suggested are applied:
```suggestion
                coupling map is unidirectional.  If there is no
                coupling map or the coupling map is bidirectional,
                the gate direction with the shorter
                duration from the backend properties will be used. If
```

I actually don't know there will ever be a situation where there is no coupling map but gate durations exist, seems contradictory. But definitely the useful situation to cover is bidirectional coupling map with asymmetric gate durations.